### PR TITLE
`ReflectionClass#isInstance()` support

### PR DESCRIPTION
**Pull request by Ocramius**
Required for #59

### CODE REVIEW

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -663,6 +664,32 @@ function (self $interface) {
     }
 
     /**
+     * Checks whether the given object is an instance
+     *
+     * @link http://php.net/manual/en/reflectionclass.isinstance.php
+     *
+     * @param object $object
+     *
+     * @throws
```
Comment (asgrim) [inline src/Reflection/ReflectionClass.php]: remove

File: test/unit/Reflection/Exception/NotAnObjectTest.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+use BetterReflection\Reflection\Exception\NotAnObject;
+use BetterReflectionTest\ClassWithInterfaces;
+use BetterReflectionTest\ClassWithInterfacesOther;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * @covers \BetterReflection\Reflection\Exception\NotAnObject
+ */
+class NotAnObjectTest extends PHPUnit_Framework_TestCase
```
Comment (asgrim) [inline test/unit/Reflection/Exception/NotAnObjectTest.php]: this test wouldn't be required if you just `throw new NotAnObjectTest('... blah ... ')` ;)

Comment (Ocramius) [inline test/unit/Reflection/Exception/NotAnObjectTest.php]: I added the test specifically to separate testing the message from the actual test that just verifies the `@throws`

Comment (asgrim) [inline test/unit/Reflection/Exception/NotAnObjectTest.php]: :+1:

**Pull request merged**